master/buildbot/test/unit/test_status_web_base.py

```diff
@@ -81,6 +83,122 @@ def test_getRequestCharset_other_params(self):
             'UTF-16')
 
 
+    ### filter_tags_by_codebases ###
+
+    def test_filter_tags_by_codebases_many_tags(self):
+        tags = ['Unstable', 'Trunk', 'Trunk-ABV', 'Trunk-Unstable', '2018.2', '2018.2-QV']
+        codebases = {'unity': '2018.2/'}
+        expected_tags = ['QV', 'Unstable']
+
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        self.assertEqual(expected_tags, filtered_tags)
+
+    def test_filter_tags_by_codebases_many_cb(self):
+        tags = ['Unstable', 'Trunk', 'Trunk-ABV', 'Trunk-Unstable', '2018.2', '2018.2-QV']
+        codebases = {'unity': 'trunk/', 'mod': '2018.2/'}
+        expected_tags = ['ABV', 'QV', 'Unstable']
+
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        self.assertEqual(expected_tags, filtered_tags)
+
+    def test_filter_tags_by_codebases_simple_unstable(self):
+        tags = ['Unstable', 'Trunk', 'Trunk-ABV', '2018.2', '2018.2-QV']
+        codebases = {'unity': 'trunk/'}
+        expected_tags = ['ABV', 'Unstable']
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        self.assertEqual(expected_tags, filtered_tags)
+
+    def test_filter_tags_by_codebases_foreign_unstable(self):
+        tags = ['Trunk', 'Trunk-ABV', '2018.2', '2018.2-QV', '2018.2-QV-Unstable']
+        codebases = {'unity': 'trunk/'}
+        expected_tags = ['ABV']
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        self.assertEqual(expected_tags, filtered_tags)
+
+    def test_filter_tags_by_codebases_empty_cb(self):
+        tags = ['Unstable', 'Trunk', 'Trunk-ABV', 'Trunk-Unstable', '2018.2', '2018.2-QV']
+        codebases = {}
+        expected_tags = sorted(tags)
+
+        filtered_tags = base.filter_tags_by_codebases(tags, codebases)
+
+        self.assertEqual(expected_tags, filtered_tags)
+
+    def test_filter_tags_by_codebases_uknown_branch(self):
+        tags = ['Unstable', 'Trunk', 'Trunk-ABV', 'Trunk-Unstable', '2018.2', '2018.2-QV']
+        codebases = {'foo': '2019.2/'}       # not unity, good pattern
```

Comment:
`unity` is not the only codebase where we use these filters, there shouldn't be any code specific for it.

Reply:
There is special case for unity in rtBuilders.js (find `MAIN_REPO= "unity_branch"`). I want to have the same behavior on builder-list and builder-page.

Reply:
ok